From: Dan Streetman <dan.streetman@canonical.com>
Subject: Fix PXE dhcp/bootp broadcast flag bit position
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=733988
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/klibc/+bug/1327412

This corrects the previous patch for the problem, as it set the wrong
bit.  The correct flags value is 0x8000, not 0x800.

diff -rupN a/usr/kinit/ipconfig/dhcp_proto.c klibc-2.0.3/usr/kinit/ipconfig/dhcp_proto.c
--- a/usr/kinit/ipconfig/dhcp_proto.c	2016-09-15 11:25:53.000000000 -0400
+++ klibc-2.0.3/usr/kinit/ipconfig/dhcp_proto.c	2016-09-15 11:28:50.919549775 -0400
@@ -208,7 +208,7 @@ static int dhcp_send(struct netdev *dev,
 	 */
 	bootp.yiaddr	= INADDR_ANY;
 	bootp.giaddr	= INADDR_ANY;
-	bootp.flags	= htons(0x800);
+	bootp.flags	= htons(0x8000);
 	bootp.secs	= htons(time(NULL) - dev->open_time);
 	memcpy(bootp.chaddr, dev->hwaddr, 16);
 
